fireworks/core/launchpad.py

```diff
@@ -583,6 +583,33 @@ def run_exists(self, fworker=None):
         q = fworker.query if fworker else {}
         return bool(self._get_a_fw_to_run(query=q, checkout=False))
 
+    def future_run_exists(self, fworker=None):
+        """Check if database has any current OR future Fireworks available
+
+        Returns:
+            bool: True if database has any ready or waiting Fireworks.
+        """
+        if self.run_exists(fworker):
+            # check first to see if any are READY
+            return True
+        else:
+            # grab all [RUNNING/RESERVED] fireworks
+            q = fworker.query if fworker else {}
+            q.update({'state': {'$in': ['RUNNING', 'RESERVED']}})
+            active = self.get_fw_ids(q)
+            # then check if they have WAITING children
+            for fw_id in active:
+                children = self.get_fw_by_id(fw_id).links[fw_id]
```

Comment:
I'm not sure if this is the best/lightest way to do this query.  I want to check the state of all children of fireworks with state of RUNNING or RESERVED

Reply:
I.e. can I access `.links` without having to deserialize the Firework